internal/relationtuple/uuid_mapping.go

```diff
@@ -0,0 +1,413 @@
+package relationtuple
+
+import (
+	"context"
+	"testing"
+
+	"github.com/ory/keto/internal/driver/config"
+	"github.com/ory/keto/internal/x"
+	"github.com/ory/keto/ketoapi"
+
+	"github.com/gofrs/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type (
+	mapperDependencies interface {
+		MappingManagerProvider
+		config.Provider
+	}
+	MappingManagerProvider interface {
+		MappingManager() MappingManager
+	}
+	MappingManager interface {
+		MapStringsToUUIDs(ctx context.Context, s ...string) ([]uuid.UUID, error)
+		MapUUIDsToStrings(ctx context.Context, u ...uuid.UUID) ([]string, error)
+	}
+	MapperProvider interface {
+		Mapper() *Mapper
+	}
+	Mapper struct {
+		D mapperDependencies
+	}
+)
+
+type success struct {
+	fs  []func()
+	err *error
+}
+
+func newSuccess(err *error) *success {
+	return &success{
+		err: err,
+	}
+}
+
+func (c *success) do(f func()) {
+	c.fs = append(c.fs, f)
+}
+
+func (c *success) cleanup() {
```

Comment:
This took me a while to understand ;). Maybe we can name it more like "apply" or "copyMappedUUIDs"? WDYT?